starky/src/stark.rs

```diff
@@ -147,79 +157,71 @@ pub trait Stark<F: RichField + Extendable<D>, const D: usize>: Sync {
         g: F,
         config: &StarkConfig,
     ) -> FriInstanceInfoTarget<D> {
-        let mut oracles = vec![];
-
-        let trace_info = FriPolynomialInfo::from_range(oracles.len(), 0..Self::COLUMNS);
-        oracles.push(FriOracleInfo {
+        let trace_oracle = FriOracleInfo {
             num_polys: Self::COLUMNS,
             blinding: false,
-        });
-
-        let permutation_zs_info = if self.uses_permutation_args() {
-            let num_z_polys = self.num_permutation_batches(config);
-            let polys = FriPolynomialInfo::from_range(oracles.len(), 0..num_z_polys);
-            oracles.push(FriOracleInfo {
-                num_polys: num_z_polys,
-                blinding: false,
-            });
-            polys
-        } else {
-            vec![]
         };
+        let trace_info = FriPolynomialInfo::from_range(TRACE_ORACLE_INDEX, 0..Self::COLUMNS);
 
-        let num_quotient_polys = self.quotient_degree_factor() * config.num_challenges;
-        let quotient_info = FriPolynomialInfo::from_range(oracles.len(), 0..num_quotient_polys);
-        oracles.push(FriOracleInfo {
+        let num_lookup_columns = self.num_lookup_helper_columns(config);
+        let num_auxiliary_polys = num_lookup_columns;
+        let auxiliary_oracle = FriOracleInfo {
+            num_polys: num_auxiliary_polys,
+            blinding: false,
+        };
+        let auxiliary_polys_info =
+            FriPolynomialInfo::from_range(AUXILIARY_ORACLE_INDEX, 0..num_auxiliary_polys);
+
+        let ctl_zs_info = FriPolynomialInfo::from_range(
+            AUXILIARY_ORACLE_INDEX,
+            num_lookup_columns..num_lookup_columns,
```

Comment:
If you keep this to prepare for the CTLs, then this should be:
```suggestion
            num_lookup_columns..num_auxiliary_polys,
```

Reply:
will remove as it's CTL related, as the other items above. Thanks for noticing!